src/api/workers/enrichment.js

```diff
@@ -1,91 +1,33 @@
-        if (!map[id]) map[id] = fastq.promise(await worker(id, ctx), 1);
-        return map[id];
-    };
-
-    return { get };
-})();
-
-const createEzsRuleCommands = rule => ezs.compileScript(rule).get();
-
-const createEnrichmentTransformerFactory = ctx => async id => {
-    const enrichment = await ctx.enrichment.findOneById(id);
-
-    return ({ id, value }) =>
-        new Promise((resolve, reject) => {
-            const input = new PassThrough({ objectMode: true });
-            const commands = createEzsRuleCommands(enrichment.rule);
-            const result = input.pipe(ezs('delegate', { commands }, {}));
-
-            result.on('data', ({ value }) => {
-                logger.info(`valueWebService : ${value}`);
-                return resolve({ value, enrichment });
-            });
-            result.on('error', error => reject({ error, enrichment }));
-
-            input.write({ id, value });
-            input.end();
+import Queue from 'bull';
+import { startEnrichmentBackground } from '../services/enrichment/enrichmentBackground';
+import repositoryMiddleware from '../services/repositoryMiddleware';
+
+export const PROCESS = 'process';
+
+export const enrichmentQueue = new Queue('enrichment', process.env.REDIS_URL, {
+    defaultJobOptions: { removeOnComplete: 100, removeOnFail: 100 },
+});
+
+enrichmentQueue.process(PROCESS, (job, done) => {
+    startEnrichment(job)
+        .then(() => {
+            job.progress(100);
+            done();
+        })
+        .catch(err => {
+            handleStartEnrichmentError();
+            done(err);
         });
-};
-
-const worker = async (id, ctx) => {
-    const transformerFactory = createEnrichmentTransformerFactory(ctx);
-
-    return async entry => {
-        const enricherTransformer = await transformerFactory(id);
-        const enrichment = await ctx.enrichment.findOneById(id);
-        await ctx.dataset.updateOne(
-            { _id: new ObjectId(entry._id) },
-            { $set: { [enrichment.name]: 'En attente' } },
-        );
+});
 
-        try {
-            let { value } = await enricherTransformer({
-                id: entry._id,
-                value: entry,
-            });
-
-            if (value === undefined) {
-                value = 'n/a';
-            }
-
-            await ctx.dataset.updateOne(
-                { _id: new ObjectId(entry._id) },
-                { $set: { [enrichment.name]: value } },
-            );
-        } catch (e) {
-            await ctx.dataset.updateOne(
-                { _id: new ObjectId(entry._id) },
-                { $set: { [enrichment.name]: `ERROR: ${e.error.message}` } },
-            );
+const startEnrichment = async job => {
+    const ctx = await prepareContext({ job });
+    await startEnrichmentBackground(ctx);
+};
 
-            logger.error('Enrichment error', {
-                enrichment,
-                error: e.error,
-            });
-        }
+const handleStartEnrichmentError = async () => {};
```

Comment:
Remove this if we don't do anything, we will add it when we need it. Or else it will stay like this ad vitam.